### PR TITLE
Rewrite project About pages to minimize API calls

### PR DESCRIPTION
**Pull request by rogerhutchings**
As pointed out by @eatyourgreens, the front end makes way too many API calls. On the project content pages (about, research etc), a call is made firstly when the about page loads, then each and every time a tab is changed. For the team page, a further call is made to get the project roles, and then again for each individual user listed.

I've rewritten these components in ES6, ditched the `PromiseRenderers`, used the [container / component model](https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0#.zcj4c6qpv) to restructure them so requests only happen in the route index, and batched calls together where I could.

I also added in counterpart to translate missing content messages, not that they should really come up.

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [x] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [x] Did you deploy a staging branch? https://about.pfe-preview.zooniverse.org/
- [ ] Did you get any type checking errors from [Babel Typecheck](https://github.com/codemix/babel-plugin-typecheck)

## Optional

- [x] If it's a new component, is it in ES6? Is it clear of warnings from ESLint?
- [x] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?

### CODE REVIEW

**Comment by rogerhutchings** [comment]
Things to do:

- When switching tabs, two more API calls get made every time, one to `/conversations` and the other to `/notifications`. They aren't coming from these components, so they should probably be tracked down and ~~killed~~ reviewed.
- I'm still using the `Avatar` component in the team page, which makes another call for each user. I wonder if we could batch together the avatar requests (not sure how they're linked to the user resource atm, other than using the `user.get('avatar')` resource method), and then pass them a URL string instead of a user object?

File: app/pages/project/about/index.jsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  }
+
+  render() {
+    return (this.state.loaded)
+      ? this.renderAbout()
+      : null;
+  }
+
+  renderAbout() {
+    const { state: { pages, team }, props: { children, project } } = this;
```
Comment (rogerhutchings) [inline app/pages/project/about/index.jsx]: destructuring like a baws

**Comment by eatyourgreens** [comment]
Looks good to me, but needs a rebase to fix merge conflicts.

**Comment by rogerhutchings** [comment]
Thanks @eatyourgreens, rebase is done so I'll merge

**Pull request merged**